File: 백준/Bronze/28295. 체육은 코딩과목 입니다/체육은 코딩과목 입니다.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        String[] dir = {"E", "S", "W", "N"};
        int cur_dir = 3;

        Scanner sc = new Scanner(System.in);

        for (int i = 0; i < 10; i++) {
            int command = sc.nextInt();
            // 1 시계방향 / 2 180도 / 3 반시계방향
            // 동(0)남(1)서(2)북(3)
            if (command == 1) {
                cur_dir = (cur_dir + 1) % 4;
            } else if (command == 2) {
                cur_dir = (cur_dir + 2) % 4;
            } else if (command == 3) {
                cur_dir = (cur_dir - 1 + 4) % 4;
            }
        }

        System.out.println(dir[cur_dir]);
    }
}

File: 백준/Bronze/28295. 체육은 코딩과목 입니다/체육은 코딩과목 입니다.py
dir = ["E", "S", "W", "N"]
cur_dir = 3

for i in range(10):
    command = int(input())
    # 1 시계방향 / 2 180도 / 3 반시계방향
    # 동(0)남(1)서(2)북(3)
    if command == 1:
        cur_dir = (cur_dir + 1) % 4
    elif command == 2:
        cur_dir = (cur_dir + 2) % 4
    elif command == 3:
        cur_dir = (cur_dir - 1 + 4) % 4

print(dir[cur_dir])